sklearn/utils/class_weight.py

```diff
@@ -69,7 +75,10 @@ def compute_class_weight(class_weight, *, classes, y):
         if not all(np.isin(classes, le.classes_)):
             raise ValueError("classes should have valid labels that are in y")
 
-        recip_freq = len(y) / (len(le.classes_) * np.bincount(y_ind).astype(np.float64))
+        weighted_class_counts = np.bincount(y_ind, weights=sample_weight)
+        recip_freq = weighted_class_counts.sum() / (
+            len(le.classes_) * weighted_class_counts.astype(np.float64)
+        )
```

Comment:
When using weights, numpy automatically converts to float64 (even with integer weights)
```suggestion
        recip_freq = weighted_class_counts.sum() / (
            len(le.classes_) * weighted_class_counts
        )
```